fcrepo-http-api/src/main/java/org/fcrepo/http/api/responses/StreamingBaseHtmlProvider.java

```diff
@@ -228,40 +196,17 @@ protected Context getContext(final Model model, final Node subject) {
     private Template getTemplate(final Model rdf, final Node subject,
                                  final List<Annotation> annotations) {
 
-        Optional<Template> template = annotations.stream()
-                                  .filter(x -> x instanceof HtmlTemplate)
-                                  .map(x -> ((HtmlTemplate) x).value())
-                                  .filter(templatesMap::containsKey)
-                                  .map(templatesMap::get)
-                                  .findFirst();
-
-        if (!template.isPresent()) {
-            LOGGER.trace("Attempting to discover mixin types of node for resource in question: {}", subject);
-            template = rdf.listObjectsOfProperty(createResource(subject.getURI()),
-                                                 createProperty(getRDFNamespaceForJcrNamespace(JCR_NAMESPACE) +
-                                                     "mixinTypes"))
-                          .toList().stream()
-                          .map(x -> x.asLiteral().getLexicalForm())
-                          .filter(templatesMap::containsKey)
-                          .map(templatesMap::get)
-                          .findFirst();
-        }
-
-        if (template.isPresent()) {
-            LOGGER.debug("Choosing template: {}", template.get().getName());
-            return template.get();
-        } else {
-            LOGGER.trace("Attempting to discover primary type of node for resource in question: {}", subject);
-            return rdf.listObjectsOfProperty(createResource(subject.getURI()),
-                                             createProperty(getRDFNamespaceForJcrNamespace(JCR_NAMESPACE) +
-                                                     "primaryType"))
-                          .toList().stream()
-                          .map(x -> x.asLiteral().getLexicalForm())
-                          .filter(templatesMap::containsKey)
-                          .map(templatesMap::get)
-                          .findFirst()
-                          .orElse(templatesMap.get("node"));
-        }
+        return annotations.stream().filter(x -> x instanceof HtmlTemplate).map(x -> ((HtmlTemplate) x).value())
+            .filter(templatesMap::containsKey).map(templatesMap::get).findFirst().orElseGet(() -> {
+                LOGGER.trace("Attempting to discover RDF types of node for resource in question: {}", subject);
+                final Template tpl = getFirst(iter(rdf.listObjectsOfProperty(createResource(subject.getURI()), type))
+                              .map(x -> x.asResource().getURI()).filter(x -> typeMap.containsKey(x))
```

Comment:
Ah, right, right. Okay, you're going to freaking shoot me, but what about `com.hp.hpl.jena.sparql.util.graph.GraphUtils.multiValueURI(Resource, Property)`? I.e. like:
```
final Resource resource = rdf.getResource(subject.getURI());
final String templateName = getFirst(multiValueURI(resource, type),"default"));
return templatesMap.get(templateName);
```